rules-tests/TypeDeclaration/Rector/FunctionLike/ReturnTypeDeclarationRector/Fixture/wrong_type_generator.php.inc

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Rector\Tests\TypeDeclaration\Rector\FunctionLike\ReturnTypeDeclarationRector\Fixture;
+
+final class WrongTypeGenerator
+{
+    /**
+     * @return \wrong
+     **/
+    public function generator()
+    {
+        yield 5;
+    }
+}
+?>
+-----
+<?php
+
+namespace Rector\Tests\TypeDeclaration\Rector\FunctionLike\ReturnTypeDeclarationRector\Fixture;
+
+final class WrongTypeGenerator
+{
+    /**
+     * @return \wrong
+     **/
+    public function generator(): \Iterator
+    {
+        yield 5;
+    }
```

Comment:
It looks like Rector handles this incorrectly. It should be`\Generator`.

https://www.php.net/manual/en/language.generators.syntax.php

Reply:
Ok, implemented to return Generator when `$functionLike->getReturnType()` is null https://github.com/rectorphp/rector-src/pull/1794/commits/b1affaee2eaceecd2760f703790b657d11b2c089